omnidata_tools/torch/Omni_own_train_val_test.py

```diff
@@ -160,33 +132,65 @@ def save_outputs(img_path, output_file_name, img_size, output_direct):
             state_dict[k[6:]] = v
     else:
         state_dict = checkpoint
+
     model.load_state_dict(state_dict)
     model.to(device)
     trans_totensor = transforms.Compose([transforms.Resize(image_size, interpolation=PIL.Image.BILINEAR),
-                                             transforms.CenterCrop(image_size),
-                                             transforms.ToTensor(),
-                                             transforms.Normalize(mean=0.5, std=0.5)])
+                                         transforms.CenterCrop(image_size),
+                                         transforms.ToTensor(),
+                                         transforms.Normalize(mean=0.5, std=0.5)])
     trans_rgb = transforms.Compose([transforms.Resize(image_size, interpolation=PIL.Image.BILINEAR),
                                     transforms.CenterCrop(image_size)])
 
-    img_path = Path(args.img_path)
     if img_path.is_file():
-        save_outputs(args.img_path, os.path.splitext(os.path.basename(args.img_path))[0], image_size, 'single')
+        save_outputs(args.img_path, os.path.splitext(os.path.basename(args.img_path))[0], image_size, 'single', trans_topil, trans_totensor, trans_rgb, device, model, 'depth')
     elif img_path.is_dir():
         for f in glob.glob(args.img_path + '/train/*rgb.png'):
-            save_outputs(f, os.path.splitext(os.path.basename(f))[0], image_size, 'train')
+            save_outputs(f, os.path.splitext(os.path.basename(f))[0], image_size, 'train', trans_topil, trans_totensor, trans_rgb, device, model, 'depth')
         for f in glob.glob(args.img_path + '/test/*rgb.png'):
-            save_outputs(f, os.path.splitext(os.path.basename(f))[0], image_size, 'test')
+            save_outputs(f, os.path.splitext(os.path.basename(f))[0], image_size, 'test', trans_topil, trans_totensor, trans_rgb, device, model, 'depth')
         for f in glob.glob(args.img_path + '/val/*rgb.png'):
-            save_outputs(f, os.path.splitext(os.path.basename(f))[0], image_size, 'val')
-    else:
-        print("invalid file path!")
-        sys.exit()
-    args.task = intermediate
+            save_outputs(f, os.path.splitext(os.path.basename(f))[0], image_size, 'val', trans_topil, trans_totensor, trans_rgb, device, model, 'depth')
```

Comment:
Here two lines are repeated 3 times with only a single change, consider putting them in a for loop

Reply:
For loop added in next version.